make/autoconf/jdk-version.m4

```diff
@@ -69,149 +69,163 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   AC_SUBST(JDK_RC_PLATFORM_NAME)
   AC_SUBST(HOTSPOT_VM_DISTRO)
 
+  # Note: UTIL_ARG_WITH treats empty strings as valid values when OPTIONAL is false!
+
   # Setup username (for use in adhoc version strings etc)
-  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
-      [build username to use in version strings])])
-  if test "x$with_build_user" = xyes || test "x$with_build_user" = xno; then
-    AC_MSG_ERROR([--with-build-user must have a value])
-  elif test "x$with_build_user" != x; then
-    USERNAME="$with_build_user"
-  else
-    # Outer [ ] to quote m4.
-    [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
-  fi
+  # Outer [ ] to quote m4.
+  UTIL_ARG_WITH(NAME: build-user, TYPE: string,
+    RESULT: USERNAME,
+    DEFAULT: [ `$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ],
+    DESC: [build username to use in version strings],
+    CHECK_VALUE: [
+      if test "x$RESULT" = x; then
+        FAILURE="--with-build-user cannot be empty"
+      fi
+    ])
   AC_SUBST(USERNAME)
 
   # Set the JDK RC name
-  AC_ARG_WITH(jdk-rc-name, [AS_HELP_STRING([--with-jdk-rc-name],
-      [Set JDK RC name. This is used for FileDescription and ProductName properties
-       of MS Windows binaries. @<:@not specified@:>@])])
-  if test "x$with_jdk_rc_name" = xyes || test "x$with_jdk_rc_name" = xno; then
-    AC_MSG_ERROR([--with-jdk-rc-name must have a value])
-  elif [ ! [[ $with_jdk_rc_name =~ ^[[:print:]]*$ ]] ]; then
-    AC_MSG_ERROR([--with-jdk-rc-name contains non-printing characters: $with_jdk_rc_name])
-  elif test "x$with_jdk_rc_name" != x; then
-    # Set JDK_RC_NAME to a custom value if '--with-jdk-rc-name' was used and is not empty.
-    JDK_RC_NAME="$with_jdk_rc_name"
-  else
-    # Otherwise calculate from "branding.conf" included above.
-    JDK_RC_NAME="$PRODUCT_NAME $JDK_RC_PLATFORM_NAME"
-  fi
+  # Otherwise calculate from "branding.conf" included above.
+  UTIL_ARG_WITH(NAME: jdk-rc-name, TYPE: string,
+    DEFAULT: $PRODUCT_NAME $JDK_RC_PLATFORM_NAME,
+    DESC: [Set JDK RC name. This is used for FileDescription and ProductName
+       properties of MS Windows binaries.],
+    DEFAULT_DESC: [not specified],
+    CHECK_VALUE: [
```

Comment:
I see that this check is repeated all over the place. I suggest that you create a new `TYPE` for this. I suggest calling it `identifier`. Then you replace `TYPE: string` with `TYPE: identifier` in all places where we check that the value is non-empty and contains only printable characters. The type itself is declared in line 692 in util.m4 (and documented on line 582). Then you need to add a type checking method:
```
AC_DEFUN([UTIL_CHECK_TYPE_identifier],
[
  # Check that the argument is non-empty and only contains printable characters
  if test "x$1" = x; then
    FAILURE="Value cannot be empty"
  elif [ ! [[ $1 =~ ^[[:print:]]*$ ]] ]; then
    FAILURE="Value contains non-printing characters: $1"
  fi
])
```

Reply:
Seems to me like a util function for CHECK_VALUE would be a slightly better approach than creating a new type, like the one we already have:

```
###############################################################################
# Helper functions for CHECK_VALUE in ARG_WITH.
AC_DEFUN([UTIL_CHECK_STRING_NON_EMPTY],
[
  if test "x$RESULT" = "x"; then
    FAILURE="Value cannot be empty"
  fi
])
```